uber/models.py

```diff
@@ -622,18 +622,11 @@ def update_badge(self, attendee, old_badge_type, old_badge_num):
             :return:
             """
             from uber.badge_funcs import needs_badge_num
-            old_badge_num = int(old_badge_num or 0) or None
-            was_dupe_num = self.query(Attendee).filter(Attendee.badge_num == old_badge_num,
-                                                       Attendee.id != attendee.id).first()
 
-            if not was_dupe_num and old_badge_type == attendee.badge_type and (not attendee.badge_num or old_badge_num == attendee.badge_num):
-                attendee.badge_num = old_badge_num
-                return 'Attendee is already {} with badge {}'.format(c.BADGES[old_badge_type], old_badge_num)
-
-            if c.SHIFT_CUSTOM_BADGES:
+            if c.SHIFT_CUSTOM_BADGES and c.BEFORE_PRINTED_BADGE_DEADLINE:
                 # fill in the gap from the old number, if applicable
                 badge_num_keep = attendee.badge_num
-                if old_badge_num and not was_dupe_num:
+                if old_badge_num:
                     self.shift_badges(old_badge_type, old_badge_num + 1, down=True)
```

Comment:
Before we were calling it from the `change_badge` badge which could theoretically be done if the admin just clicked 'save' without changing anything. However, this is now only called from the presave adjustments if `self.orig_value_of('badge_type') != self.badge_type or self.orig_value_of('badge_num') != self.badge_num`, so this is never the same badge.